storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupUtils.java

```diff
@@ -0,0 +1,128 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.container.cgroup;
+
+import org.apache.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class CgroupUtils {
+
+    public static final String CGROUP_STATUS_FILE = "/proc/cgroups";
+    public static final String MOUNT_STATUS_FILE = "/proc/mounts";
+
+    private static final Logger LOG = LoggerFactory.getLogger(CgroupUtils.class);
+
+    public static void deleteDir(String dir) {
+        File d = new File(dir);
+        if (d.exists()) {
+            if (d.isDirectory()) {
+                if (!d.delete()) {
+                    throw new RuntimeException("Cannot delete dir " + dir);
+                }
+            } else {
+                throw new RuntimeException("dir " + dir + " is not a directory!");
+            }
+        } else {
+            LOG.warn("dir {} does not exist!", dir);
+        }
+    }
+
+    /**
+     * Get a set of SubSystemType objects from a comma delimited list of subsystem names
+     */
+    public static Set<SubSystemType> getSubSystemsFromString(String str) {
+        Set<SubSystemType> result = new HashSet<SubSystemType>();
+        String[] subSystems = str.split(",");
+        for (String subSystem : subSystems) {
+            SubSystemType type = SubSystemType.getSubSystem(subSystem);
+            if (type != null) {
+                result.add(type);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Get a string that is a comma delimited list of subsystems
+     */
+    public static String subSystemsToString(Set<SubSystemType> subSystems) {
+        StringBuilder sb = new StringBuilder();
+        if (subSystems.size() == 0) {
+            return sb.toString();
+        }
+        for (SubSystemType type : subSystems) {
+            sb.append(type.name()).append(",");
+        }
+        return sb.toString().substring(0, sb.length() - 1);
+    }
+
+    public static boolean enabled() {
+        return Utils.checkFileExists(CGROUP_STATUS_FILE);
+    }
+
+    public static List<String> readFileByLine(String fileDir) throws IOException {
```

Comment:
readFileByLine can be replaced with Files.readlines(new File(fileDir)) available in guava and already added as dependency